query/execute: data race in tests

This reproduces instantly on my machine. We probably would have noticed this earlier if CI was running tests with -race.

```
$ go test -race -run=TestExecutor_Execute -count=50 ./query/execute/...
==================
WARNING: DATA RACE
Read at 0x00c00001ad20 by goroutine 73:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:33 +0x55
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:93 +0x5c
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c00001ad20 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:54 +0x265
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
==================
WARNING: DATA RACE
Read at 0x00c000195920 by goroutine 73:
  github.com/influxdata/platform/query/execute.(*groupKey).Cols()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:24 +0x3a
  github.com/influxdata/platform/query/execute.AddTableKeyCols()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/table.go:89 +0x42
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:98 +0x135
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c000195920 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:17 +0x1db
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
==================
WARNING: DATA RACE
Read at 0x00c0001958f0 by goroutine 73:
  github.com/influxdata/platform/query/execute.AddTableKeyCols()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/table.go:89 +0x87
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:98 +0x135
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c0001958f0 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:44 +0x45c
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
==================
WARNING: DATA RACE
Read at 0x00c000195938 by goroutine 73:
  github.com/influxdata/platform/query/execute.(*groupKey).ValueTime()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:57 +0x42
  github.com/influxdata/platform/query/execute.AppendAggregateTime()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:226 +0x2e4
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:147 +0x735
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c000195938 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:17 +0x225
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
==================
WARNING: DATA RACE
Read at 0x00c00032e9d0 by goroutine 73:
  github.com/influxdata/platform/query/execute.(*groupKey).ValueTime()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:57 +0x74
  github.com/influxdata/platform/query/execute.AppendAggregateTime()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:226 +0x2e4
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:147 +0x735
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c00032e9d0 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:52 +0x5fb
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
==================
WARNING: DATA RACE
Read at 0x00c00032e9c0 by goroutine 73:
  github.com/influxdata/platform/query/execute.(*groupKey).ValueTime()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/group_key.go:57 +0x74
  github.com/influxdata/platform/query/execute.AppendKeyValues()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/table.go:241 +0x276
  github.com/influxdata/platform/query/execute.(*aggregateTransformation).Process()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/aggregate.go:192 +0x9db
  github.com/influxdata/platform/query/execute.processMessage()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:199 +0x257
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:156 +0x107
  github.com/influxdata/platform/query/execute.(*consecutiveTransport).processMessages-fm()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/transport.go:137 +0x4b
  github.com/influxdata/platform/query/execute.(*poolDispatcher).run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:126 +0x7a
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:80 +0xa2

Previous write at 0x00c00032e9c0 by goroutine 70:
  github.com/influxdata/platform/query/execute/executetest.(*Table).Normalize()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:52 +0x5fb
  github.com/influxdata/platform/query/execute/executetest.(*Table).Key()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executetest/table.go:69 +0x38
  github.com/influxdata/platform/query/execute_test.(*testFromProcedureSource).Run()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:592 +0x2c1
  github.com/influxdata/platform/query/execute.(*executionState).do.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:224 +0x7c

Goroutine 73 (running) created at:
  github.com/influxdata/platform/query/execute.(*poolDispatcher).Start()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/dispatcher.go:61 +0x8b
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:227 +0x143
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162

Goroutine 70 (finished) created at:
  github.com/influxdata/platform/query/execute.(*executionState).do()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:205 +0xce
  github.com/influxdata/platform/query/execute.(*executor).Execute()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor.go:77 +0x150
  github.com/influxdata/platform/query/execute_test.TestExecutor_Execute.func1()
      /Users/mr/go/src/github.com/influxdata/platform/query/execute/executor_test.go:535 +0x1c1
  testing.tRunner()
      /usr/local/Cellar/go/1.11rc1/libexec/src/testing/testing.go:827 +0x162
==================
--- FAIL: TestExecutor_Execute (0.01s)
    --- FAIL: TestExecutor_Execute/multiple_aggregates (0.00s)
        testing.go:771: race detected during execution of test
    testing.go:771: race detected during execution of test
FAIL
FAIL	github.com/influxdata/platform/query/execute	0.839s
?   	github.com/influxdata/platform/query/execute/executetest	[no test files]
```

--- a/query/execute/executor_test.go
+++ b/query/execute/executor_test.go
@@ -44,8 +44,8 @@ func TestExecutor_Execute(t *testing.T) {
 				Procedures: map[plan.ProcedureID]*plan.Procedure{
 					plan.ProcedureIDFromOperationID("from"): {
 						ID: plan.ProcedureIDFromOperationID("from"),
-						Spec: &testFromProcedureSource{
-							data: []query.Table{&executetest.Table{
+						Spec: newTestFromProcedureSource(
+							[]*executetest.Table{&executetest.Table{
 								KeyCols: []string{"_start", "_stop"},
 								ColMeta: []query.ColMeta{
 									{Label: "_start", Type: query.TTime},
@@ -61,7 +61,7 @@ func TestExecutor_Execute(t *testing.T) {
 									{execute.Time(0), execute.Time(5), execute.Time(4), 5.0},
 								},
 							}},
-						},
+						),
 						Bounds: &plan.BoundsSpec{
 							Start: values.ConvertTime(time.Unix(0, 1)),
 							Stop:  values.ConvertTime(time.Unix(0, 5)),
@@ -114,8 +114,8 @@ func TestExecutor_Execute(t *testing.T) {
 				Procedures: map[plan.ProcedureID]*plan.Procedure{
 					plan.ProcedureIDFromOperationID("from"): {
 						ID: plan.ProcedureIDFromOperationID("from"),
-						Spec: &testFromProcedureSource{
-							data: []query.Table{&executetest.Table{
+						Spec: newTestFromProcedureSource(
+							[]*executetest.Table{&executetest.Table{
 								KeyCols: []string{"_start", "_stop"},
 								ColMeta: []query.ColMeta{
 									{Label: "_start", Type: query.TTime},
@@ -131,7 +131,7 @@ func TestExecutor_Execute(t *testing.T) {
 									{execute.Time(0), execute.Time(5), execute.Time(4), int64(5)},
 								},
 							}},
-						},
+						),
 						Bounds: &plan.BoundsSpec{
 							Start: values.ConvertTime(time.Unix(0, 1)),
 							Stop:  values.ConvertTime(time.Unix(0, 5)),
@@ -218,8 +218,8 @@ func TestExecutor_Execute(t *testing.T) {
 				Procedures: map[plan.ProcedureID]*plan.Procedure{
 					plan.ProcedureIDFromOperationID("from"): {
 						ID: plan.ProcedureIDFromOperationID("from"),
-						Spec: &testFromProcedureSource{
-							data: []query.Table{
+						Spec: newTestFromProcedureSource(
+							[]*executetest.Table{
 								&executetest.Table{
 									KeyCols: []string{"_start", "_stop", "_key"},
 									ColMeta: []query.ColMeta{
@@ -286,7 +286,7 @@ func TestExecutor_Execute(t *testing.T) {
 									},
 								},
 							},
-						},
+						),
 						Parents: nil,
 						Bounds: &plan.BoundsSpec{
 							Start: values.ConvertTime(time.Unix(0, 1)),
@@ -437,8 +437,8 @@ func TestExecutor_Execute(t *testing.T) {
 				Procedures: map[plan.ProcedureID]*plan.Procedure{
 					plan.ProcedureIDFromOperationID("from"): {
 						ID: plan.ProcedureIDFromOperationID("from"),
-						Spec: &testFromProcedureSource{
-							data: []query.Table{&executetest.Table{
+						Spec: newTestFromProcedureSource(
+							[]*executetest.Table{&executetest.Table{
 								KeyCols: []string{"_start", "_stop"},
 								ColMeta: []query.ColMeta{
 									{Label: "_start", Type: query.TTime},
@@ -454,7 +454,7 @@ func TestExecutor_Execute(t *testing.T) {
 									{execute.Time(0), execute.Time(5), execute.Time(4), 5.0},
 								},
 							}},
-						},
+						),
 						Bounds: &plan.BoundsSpec{
 							Start: values.ConvertTime(time.Unix(0, 1)),
 							Stop:  values.ConvertTime(time.Unix(0, 5)),
@@ -565,8 +565,19 @@ func TestExecutor_Execute(t *testing.T) {
 }
 
 type testFromProcedureSource struct {
-	data []query.Table
+	data []*executetest.Table
 	ts   []execute.Transformation
+}
+
+func newTestFromProcedureSource(data []*executetest.Table) *testFromProcedureSource {
+	p := &testFromProcedureSource{
+		data: data,
+	}
+	// Normalize the data before anything can read it
+	for _, tbl := range p.data {
+		tbl.Normalize()
+	}
+	return p
 }
 
 func (p *testFromProcedureSource) Kind() plan.ProcedureKind {
@@ -585,11 +596,11 @@ func (p *testFromProcedureSource) Run(ctx context.Context) {
 	id := execute.DatasetID(uuid.NewV4())
 	for _, t := range p.ts {
 		var max execute.Time
-		for _, b := range p.data {
-			t.Process(id, b)
-			stopIdx := execute.ColIdx(execute.DefaultStopColLabel, b.Cols())
+		for _, tbl := range p.data {
+			t.Process(id, tbl)
+			stopIdx := execute.ColIdx(execute.DefaultStopColLabel, tbl.Cols())
 			if stopIdx >= 0 {
-				if s := b.Key().ValueTime(stopIdx); s > max {
+				if s := tbl.Key().ValueTime(stopIdx); s > max {
 					max = s
 				}
 			}

